tests/test_cache.py

```diff
@@ -481,23 +448,26 @@ def test_remove_expired_entries(self, datetime_mock, datetime_mock2):
 
     def test_cache_unpickle_errors(self):
         url = httpbin('get?q=1')
-        self.assertFalse(self.s.get(url).from_cache)
-        with mock.patch("requests_cache.backends.base.BaseCache.restore_response", side_effect=TypeError):
+        assert self.s.get(url).from_cache is False
+
+        # If there is an error during deserialization, the request should be made again
+        with patch.object(DbPickleDict, '__getitem__', side_effect=PickleError):
             resp = self.s.get(url)
-            self.assertFalse(resp.from_cache)
-            self.assertEqual(resp.json()["args"]["q"], "1")
+            assert resp.from_cache is False
+            assert resp.json()["args"]["q"] == "1"
+
         resp = self.s.get(url)
-        self.assertTrue(resp.from_cache)
-        self.assertEqual(resp.json()["args"]["q"], "1")
+        assert resp.from_cache is True
```

Comment:
`assert resp.from_cache` should be sufficient.

What is the general idea of going back to assert instead of self.assert*? Just curious, as I tried to use the latter :)

Reply:
Good question. I only just started with some of these tests, but I plan to rewrite most of these tests as pytest-style functions.

The benefit of the `self.assert*` methods vs just `assert` is customized test output when a test fails. With pytest, all `assert` statements get customized output ([more info here](https://docs.pytest.org/en/stable/assert.html). `unittest.TestCase` was modeled after the xunit frameworks from other languages like Java. Pytest does the same things but a bit more pythonic.

My favorite is [fixtures](https://docs.pytest.org/en/stable/fixture.html). Instead of `setUp` and `tearDown`, you can write setup/teardown logic as context managers, share them across modules, [parametreize](https://docs.pytest.org/en/stable/fixture.html#parametrizing-fixtures) them, cusomize the [scope](https://docs.pytest.org/en/stable/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session), [autouse](https://docs.pytest.org/en/stable/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request) them to apply to everything in a module or session, and lots of other stuff. You can see an example [here](https://github.com/reclosedev/requests-cache/blob/26d11a674831d6046d62ee101dda85493c163f92/tests/conftest.py) that makes requests-cache work with requests-mock, using a sqlite db that gets created and deleted after every test function.

Reply:
P.S., thanks for reviewing! I'll ping you again when I'm done with the tests, if you wouldn't mind looking them over again.

Reply:
P.P.S, in general, new unit tests in PRs can be written in whatever style the author wants. `TestCase` classes are just fine. More tests are always better than no tests!

In this case I am rewriting tests for `CachedSession` and `BaseCache` because I made a large number of changes to those classes, and I want to be sure I understand what all the tests are supposed to be doing so I'm less likely to break stuff!

Reply:
@shoeffner I think I'm mostly done messing with [test_cache.py](https://github.com/reclosedev/requests-cache/blob/6ef027a423fc0661831f5f34b33c9e4972841136/tests/test_cache.py), if you want to take another look. There are a few TODOs in there that I'll handle in a separate PR to finish up #169.

Reply:
Thanks for the writeup on pytest!
I will take a look at the tests now.